### PR TITLE
Formatting Prometheus metrics output

### PR DESCRIPTION
**Pull request by konradzdeb**
# Key:

 * metrics for `query` and `rangeQuery` are now formatted and returned as clean data frames.
 * Travis will check across various R versions and macOS

# Minor:

 * `README` shows both examples
 * Documentation fixes
 * Utility functions documented together and saved in across separate files when needed

# Outstanding

* More tests for formatting of the `metrics` object
* Refac on metrics formatting
* Some missing documentation and test structure
image

## Suggested

* Richer package description
* `as.data.frame` and `summary` methods for core class
* Appveyor tests for MS Windows machine ([krlmlr/r-appveyor](https://github.com/krlmlr/r-appveyor))

### CODE REVIEW

File: R/format_metrics_data.R [hunk 1]
```diff
@@ -31,5 +33,43 @@ format_metrics_current_data <- function(x) {
...
+    }
+  ) -> dfs_to_bind
+
+  i <- 1
+  Reduce(rbind,
+         lapply(
+           X = dfs_to_bind,
+           FUN = function(dfs) {
+             suppressWarnings(cbind(x_metrics[i,], dfs)) -> x
+             i <<- i + 1
```
Comment (konradzdeb) [inline R/format_metrics_data.R:69]: @glenn-m This feels weird and I would change it at some point. At the moment, didn't have ideas on how to do it better without falling on additional packages and I didn't want to add dependencies only for that.

Comment (glenn-m) [reply R/format_metrics_data.R:69]: I'm in the process of reading through this, but does `bind(x_metrics[i,], dfs, row.names = NULL)` not resolve this?

**Comment by konradzdeb** [comment]
To remove warnings? I think the warning was about different number of rows as the one data frame has one and the other (values with that time stamp) has 7...

--
Konrad

**Comment by glenn-m** [comment]
Merged, thanks for contributing this, makes a huge improvement :+1:

**Comment by konradzdeb** [comment]
Hi, no problem it’s a good project. Maybe we can get it on CRAN some day?

--
Konrad

**Pull request merged**